static/js/src/advantage/subscribe/react/components/BuyButton/BuyButton.tsx

```diff
@@ -54,13 +30,85 @@ function StepOne({ setStep, error, setError }: StepOneProps) {
     isLoading: isPendingPurchaseLoading,
   } = usePendingPurchase();
 
+  const { product, quantity } = useProduct();
   const GAFriendlyProduct = {
     id: product?.id,
     name: product?.name,
     price: product?.price?.value / 100,
     quantity: quantity,
   };
 
+  const onStartTrialClick = () => {
```

Comment:
Ideally, this entire function should be covered by tests. It's tedious to QA this manually and easy to break accidentally.

Reply:
I've added e2e tests for both purchase and free trial